### ISSUE TITLE
slidesToShow, infinite, centerMode has unexpected behaviour

### ISSUE THREAD

**Issue by Sailias** [akiran/react-slick#1251]
https://codesandbox.io/s/24lj240mkr

In centerMode I would expect the last slide to still go to the center.
You need to short circuit the centerMode check:

```javascript
// whether or not we can go next
var canGoNext = exports.canGoNext = function canGoNext(spec) {
  var canGo = true;
  if (!spec.infinite) {
    if(spec.centerMode) {
      if (spec.currentSlide >= spec.slideCount - 1) {
        canGo = false;
      }
    }
    else if (spec.slideCount <= spec.slidesToShow || spec.currentSlide >= spec.slideCount - spec.slidesToShow) {
      canGo = false;
    }
  }
  return canGo;
};
```

**Comment by dm-randy** [comment]
I believe this should be fixed by #1232.

**Comment by randallreedjr** [comment]
I believe this should be fixed by #1232.

**Comment by randallreedjr** [comment]
Also, this issue appears to be a duplicate of #1230.
